models/parse_util.go

```diff
@@ -0,0 +1,62 @@
+package models
+
+import (
+	"fmt"
+)
+
+func normalizeMapStringInterface(source map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
NOTE<not sure>: what about: `jsonMarshallable/jsonify` the other could be `recursiveJSONMarshallable/JSONify`, since the point of these methods is to make the type json compatible/marshallable